### ISSUE TITLE
Error in sample_lsm code

### ISSUE THREAD

**Issue by maryjben** [r-spatialecology/landscapemetrics#312]
I am completing a project where I would like to sample landscape edge density of a raster within 60 meters of sampling points. My raster has two primary values, 1 and 2, and 0 is NoData and has a pixel size of 30x30 m. Both the raster and points are using the same CRS, which is NAD83(2011). 
My initial code was 
`sample_lsm(raster_awei, y = pt, size = 60, what = "lsm_l_ed")`
But I received this error.

Error: [crop] invalid extent.

I assumed this meant that the raster and points did not have the same extent, which was true, as shown below. 

```
> ext(raster_awei)
SpatExtent : 319159.3442, 881659.3442, 9587.95569999958, 739397.9557 (xmin, xmax, ymin, ymax)
> st_bbox(pt)
    xmin     ymin     xmax     ymax 
461943.0 115493.7 779919.3 612538.4 
```
I then cropped the raster to match the point extent.

```
 awei_crop = crop(raster_awei,bbox_points)
> ext(awei_crop)
SpatExtent : 461929.3442, 779929.3442, 115487.9557, 612527.9557 (xmin, xmax, ymin, ymax)
```
However, I continue to receive the same error when running the sample_lsm function. Could it be the radius or is the extent still the problem? I have also included screenshots of the plots of my original raster, cropped raster, and points on the cropped raster. Let me know if there is any other information that would be helpful. 

<img width="730" alt="Raster" src="https://github.com/r-spatialecology/landscapemetrics/assets/152836093/8e67bac2-50e4-477f-9dba-07c6570193a7">
<img width="646" alt="Cropped raster" src="https://github.com/r-spatialecology/landscapemetrics/assets/152836093/de88f8c1-0650-4389-b5c1-15001f3d39e9">
<img width="620" alt="points" src="https://github.com/r-spatialecology/landscapemetrics/assets/152836093/afb35753-a242-4506-a313-c3b725df9b82">

**Comment by mhesselbarth** [comment]
Hey, 

Sharing the data would be extremely useful yes. I dont think the raster and the points dont need exactly the same extent, as long as the points overlap the raster to a degree. 

But again, having a (subset) of the data to reproduce the error would be helpful

**Comment by maryjben** [comment]
I have included a zipped file of the data. It includes the output of an exported raster as a GEOTIFF from ArcGIS Pro and the exported points as a shapefile also from ArcGIS Pro. The points were projected to use the same CRS in ArcGIS Pro. 

This is the first time I am using R to do spatial analysis. I have previously only used ArcGIS Pro, so I apologize for my lack of experience. Especially if this is a simple fix. Also, I am using R version 4.3.2. and landscapemetrics 2.1.0 version is loaded. I am using a MacOS and have recently updated Xquartz.

[lsm_data.zip](https://github.com/r-spatialecology/landscapemetrics/files/13548638/lsm_data.zip)

**Comment by Nowosad** [comment]
@mhesselbarth see the attached code -- it has a spurious warning ("#> Warning: [crop] CRS do not match") -- can you remove it internally? Btw I was unable to find an error on a sample of the points.

@maryjben It seems that you have a lot of points (more than a million). My rough calculations show that it may take about two weeks to complete this operation...

``` r
# remotes::install_github("r-spatialecology/landscapemetrics")
library(landscapemetrics)
library(terra)
#> terra 1.7.55
library(sf)
#> Linking to GEOS 3.12.0, GDAL 3.7.2, PROJ 9.2.1; sf_use_s2() is TRUE

raster_awei = rast("Reclass_AWEI.tif")
pt = read_sf("points_project.shp")

st = system.time({sl <- sample_lsm(raster_awei, y = pt[1:100, ], size = 60, what = "lsm_l_ed")})
#> Warning: [crop] CRS do not match
sl
#> # A tibble: 100 × 8
#>    layer level     class    id metric value plot_id percentage_inside
#>    <int> <chr>     <int> <int> <chr>  <dbl>   <int>             <dbl>
#>  1     1 landscape    NA    NA ed     104.        1              100.
#>  2     1 landscape    NA    NA ed       0         2              100 
#>  3     1 landscape    NA    NA ed       0         3              100 
#>  4     1 landscape    NA    NA ed     188.        4              100 
#>  5     1 landscape    NA    NA ed      83.3       5              100 
#>  6     1 landscape    NA    NA ed       0         6              100.
#>  7     1 landscape    NA    NA ed       0         7              100.
#>  8     1 landscape    NA    NA ed     125         8              100 
#>  9     1 landscape    NA    NA ed     125         9              100 
#> 10     1 landscape    NA    NA ed      62.5      10              100 
#> # ℹ 90 more rows

st
#>    user  system elapsed 
#>   4.415   0.012   4.473

st[[3]] * (nrow(pt)/100)/3600
#> [1] 13.01959
```

<sup>Created on 2023-12-04 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by Nowosad** [comment]
@maryjben thank for the explanation. The .shp file you provided has 108 (non-empty) points. I was able to run it without an error (but with a spurious warning). 
Could you show me how have you read the raster and vector files into R?

``` r
# remotes::install_github("r-spatialecology/landscapemetrics")
library(landscapemetrics)
library(terra)
#> terra 1.7.55
library(sf)
#> Linking to GEOS 3.12.0, GDAL 3.7.2, PROJ 9.2.1; sf_use_s2() is TRUE

raster_awei = rast("Reclass_AWEI.tif")
pt = read_sf("points_project.shp")
pt = pt[!st_is_empty(pt), ]

st = system.time({sl <- sample_lsm(raster_awei, y = pt, size = 60, what = "lsm_l_ed")})
#> Warning: [crop] CRS do not match
sl
#> # A tibble: 109 × 8
#>    layer level     class    id metric value plot_id percentage_inside
#>    <int> <chr>     <int> <int> <chr>  <dbl>   <int>             <dbl>
#>  1     1 landscape    NA    NA ed     104.        1              100.
#>  2     1 landscape    NA    NA ed       0         2              100 
#>  3     1 landscape    NA    NA ed       0         3              100 
#>  4     1 landscape    NA    NA ed     188.        4              100 
#>  5     1 landscape    NA    NA ed      83.3       5              100 
#>  6     1 landscape    NA    NA ed       0         6              100.
#>  7     1 landscape    NA    NA ed       0         7              100.
#>  8     1 landscape    NA    NA ed     125         8              100 
#>  9     1 landscape    NA    NA ed     125         9              100 
#> 10     1 landscape    NA    NA ed      62.5      10              100 
#> # ℹ 99 more rows
```

<sup>Created on 2023-12-04 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by maryjben** [comment]
108 points is correct. I have read in the files as...
```
raster_awei = rast("Reclass_AWEI.tif")
pt <- st_read("points_project.shp")
```
Is the problem the "st_read"? Also, I thought I had projected the feature class in GIS to use the same CRS as the raster, but I guess it is not reading it that way.

**Comment by Nowosad** [comment]
@maryjben you can ignore this `Warning: [crop] CRS do not match` warning -- this is a bug in the code that we need to fix. Your data CRS seems to be fine. 

Also -- as I showed above -- I was able to get the output for the calculations...

**Comment by mhesselbarth** [comment]
So we only need to suppress the warning? I will have a look tomorrow morning first thing.

**Comment by mhesselbarth** [comment]
I looked a bit further into this. The issue was actually that we lost the CRS information during `construct_buffer` which is used internally to transform the points into sampling polygons. Should be fixed now with 2ae9d209  .
